### ISSUE TITLE
When Tooltip is used with a child chart, and it receives new props, the child chart never updates

### ISSUE THREAD

**Issue by jchen-eb** [eventbrite/britecharts-react#116]
## Expected Behavior
The child chart should update when props are swapped

## Current Behavior
The child chart does not update

## Steps to Reproduce (for bugs)

1. Create a chart with the Tooltip parent and a child chart (ex. StackedArea), with a valid data prop
2. Pass a new valid data prop to the child chart
3. Notice that child chart does not re-render with new data

## Context
The reason this happens is because we are rendering the child chart in Tooltip's `constructor` function, which only gets run on instantiation.

Even passing new props directly to the child chart does not seem to help.

We probably need to refactor the way the Tooltip interacts with its child chart.

## Your Environment

* Version used:
* Browser Name and version:
* Operating System and version:
* Link to your project or codepen:

**Comment by ronvenna** [comment]
Bump. Any workaround for this currently?

**Comment by dalerasrorov-eb** [comment]
@ronvenna thanks for bumping, we have not yet looked in this issue (due to different project priorities). I will look into it some time. Feel free to make a PR if you find a fix as well!

**Comment by ronvenna** [comment]
@DalerAsrorov no problem, thanks for the quick response. Will take a look at this and submit a PR.

**Comment by jchen-eb** [comment]
Hi @ronvenna,

One workaround I've been able to use is to wrap the entire chart in a functional component. This way, it forces React to instantiate a new object every time it gets new props.

So instead of 
```
render() {
	const renderStackedArea = (props) => (
            <ResponsiveStackedArea
                data={stackedAreaData.with3Sources()}
                {...props}
            />
        );

	return (
            <Tooltip
                data={stackedAreaData.with3Sources()}
                render={renderStackedArea}
            />
	);
}
```

you can do:

```
render() {
	const renderStackedArea = (props) => (
            <ResponsiveStackedArea
                data={props.data}
                {...props}
            />
        );

	const TooltipWithStackedArea = ({data}) => (
		<Tooltip
		    data={data}
		    render={renderStackedArea}
		/>
	);

	return (
            <TooltipWithStackedArea
        	data={stackedAreaData.with3Sources()}
            />
	);
}
```

This way, every time `render()` gets called, it creates a new functional component which it then renders, creating a new instance of the chart with the new data and replacing the old.

In theory, this will result in worse performance. However, in my testing I haven't noticed it being any worse on the various devices I've tried.

**Comment by davegomez** [comment]
This is an issue that is affecting other charts as well, so I created a [prototype](https://codesandbox.io/s/43101lzyw9) comparing the **Bar** and **Line** charts where you can see the issue happening.

I can work on a solution for this and make a PR but I need some help finding the issue.

**Comment by davegomez** [comment]
This is already fixed too in the PR #164

**Comment by miglesiasEB** [comment]
Thanks!
